### PR TITLE
feat(flutter): Add push notification docs

### PR DESCRIPTION
**Pull request by cshfang**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by cshfang** [comment]
src/fragments/lib/push-notifications/react-native/app_badge_count/app-badge-count.mdx ->
src/fragments/lib/push-notifications/common/app-badge-count.mdx

File: src/fragments/lib/push-notifications/flutter/request_permissions/request-permissions.mdx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+```dart
+final result = await Amplify.Notifications.Push.requestPermissions(
+  // permissions are true by default
+  // alert: true
```
Comment (HuiSF) [inline src/fragments/lib/push-notifications/flutter/request_permissions/request-permissions.mdx:4]: Intended to comment out this line?

File: src/fragments/lib/push-notifications/common/enable-rich-notifications.mdx [hunk 2]
```diff
@@ -33,13 +33,13 @@ target 'MyNotificationServiceExtension' do # Replace with your service extension
 end
 ```
 
-3. Install the new pods by running the following command at the root of your React Native project.
+3. Install the new pods by running the following command at the root of your application project.
 
 ```bash
 npx pod-install
```
Comment (HuiSF) [inline src/fragments/lib/push-notifications/common/enable-rich-notifications.mdx:39]: Might need a flutter fragment 😅 

```bash
pod install
```

Also need to add a note about installing cocoapods to make `pod` command available?

File: src/fragments/lib/push-notifications/common/request-permissions.mdx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+To learn more about the platform-specific guidances for requesting permissions, you can visit the respective documentations for [iOS](https://developer.apple.com/documentation/usernotifications/asking_permission_to_use_notifications) and [Android](https://developer.android.com/develop/ui/views/notifications/notification-permission). To best aid you in giving your users a good permission experience with platform idiomatic flows, Amplify provides the functionality below.
+
+## Get permission status
+
+The first step to request permissions from your user is to understand the current status of permissions. Your app may behave differently in response to these possible statuses below.
+
+- **Should Request** - No permissions have been requested yet. It is idiomatic at this time to simply request for permissions from the user.
+- **Should Explain Then Request** - It is recommended at this time to provide some context or rationale to the user explaining why you are requesting permission to display notifications to them before then requesting for permissions.
```
Comment (Samaritan1011001) [inline src/fragments/lib/push-notifications/common/request-permissions.mdx:10]:
```suggestion
- **Should Explain Then Request** - It is recommended at this time to provide some context or rationale to the user explaining why you want to send them push notifications before requesting for permissions.
```

Comment (abdallahshaban557) [reply src/fragments/lib/push-notifications/common/request-permissions.mdx:10]: Agreed with this as well ^

Comment (abdallahshaban557) [reply src/fragments/lib/push-notifications/common/request-permissions.mdx:10]: @cshfang - wanna accept this so that I can approve the PR?

**Review by HuiSF — DISMISSED** [review]

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by HuiSF — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**